### ISSUE TITLE
Medium Editor is... not great

### ISSUE THREAD

**Issue by lowercasename** [lowercasename/sweet#71]
I don't think we should write our own editor, because that's ridiculous, that's why. I think we can give up on looking for an editor with inline text formatting tools and go for one a lot of people have worked on for a long time, maybe one of these:

- https://prosemirror.net/ (I think ProseMirror is mature and quite complex - Nick Santos of Medium recommends it in [this very interesting piece on how difficult web WYIWYG editing is](https://medium.engineering/why-contenteditable-is-terrible-122d8a40e480) (Medium now does actually use contenteditable lol)
- https://quilljs.com/
- https://trix-editor.org/
- https://alex-d.github.io/Trumbowyg/

**Comment by toBeOfUse** [comment]
I think Medium always used contenteditable, the article you linked to talks about using contenteditable for user input and display, but also building a more straightforward model of the document that the user's creating behind the scenes, and reconstructing the content of the contenteditable from that model when necessary

**Comment by lowercasename** [comment]
Further to this, Quill has a 'bubble' mode like medium-editor and their example works well on iOS.

**Comment by lowercasename** [comment]
Ooh, also: https://docs.slatejs.org/

**Comment by toBeOfUse** [comment]
I nominate Quill, the others I think are more complex than we need. The setting it up on the page process is also almost identical, the only difference I can see is that we'll have to iterate through the elements we want to attach it to ourselves, it doesn't automatically attach to more than one at a time. The only weird thing is that it doesn't seem to provide an API method to get the content that the user has just entered in the form of HTML DOM elements? It wants to give us the user's document in its "Delta" JSON format, looks like this:
![image](https://user-images.githubusercontent.com/49729978/60684644-f5435d00-9e6c-11e9-9a1f-c0c7d41e704d.png)
Which wouldn't be too hard to iterate over and turn to HTML within our own code? But also, obviously Quill still constructs a version of the user's input in the DOM, that's what's displayed to the user to look at as they're typing stuff, we can just find that in the page with a selector and then call .html() on it to get the contents as a string, and submit that to the server. Still slightly weird that that's not explicitly exposed functionality, though, but as long as we never, like, update to a different Quill version that would require a different selector to get the HTML of what the user wrote we should be fine.

**Comment by ArloJamesBarnes** [comment]
I don't know why, but today for me Firefox and Chrome both started refusing to accept third-party content from cdn.quilljs.com, even after I added that subdomain, the root domain, and sweet.sh to security exception lists. So when I click on the text box, the browser acts like it doesn't have the texteditable class or something.

**Comment by lowercasename** [comment]
That's super weird as heck. So you're completely locked out of posting? It still works for me. I'm currently coding up a brand new editor to replace it, but that's a few weeks away at the moment. I can see if maybe there's a new version of Quill available?

**Comment by ArloJamesBarnes** [comment]
Yeah. I can still 🤝 so I'll satisfy myself with that until the new editor is in place.
